components/Chips/src/MDCChipView.m

```diff
@@ -0,0 +1,626 @@
+  }
+  if (borderWidth != nil) {
+    return (CGFloat)borderWidth.doubleValue;
+  }
+  return 0;
+}
+
+- (void)setBorderWidth:(CGFloat)borderWidth forState:(UIControlState)state {
+  _borderWidths[@(state)] = @(borderWidth);
+
+  [self updateBorderWidth];
+}
+
+- (void)updateBorderWidth {
+  self.layer.borderWidth = [self borderWidthForState:self.state];
+}
+
+- (CGFloat)elevationForState:(UIControlState)state {
+  NSNumber *elevation = _elevations[@(state)];
+  if (elevation == nil && state != UIControlStateNormal) {
+    elevation = _elevations[@(UIControlStateNormal)];
+  }
+  if (elevation != nil) {
+    return (CGFloat)[elevation doubleValue];
+  }
+  return 0;
+}
+
+- (void)setElevation:(CGFloat)elevation forState:(UIControlState)state {
+  _elevations[@(state)] = @(elevation);
+
+  [self updateElevation];
+}
+
+- (void)updateElevation {
+  CGFloat newElevation = [self elevationForState:self.state];
+  if (self.layer.elevation != newElevation) {
+    self.layer.elevation = newElevation;
+  }
+}
+
+- (nullable UIColor *)titleColorForState:(UIControlState)state {
+  UIColor *titleColor = _titleColors[@(state)];
+  if (!titleColor && state != UIControlStateNormal) {
+    titleColor = _titleColors[@(UIControlStateNormal)];
+  }
+  return titleColor;
+}
+
+- (void)setTitleColor:(nullable UIColor *)titleColor forState:(UIControlState)state {
+  _titleColors[@(state)] = titleColor;
+
+  [self updateTitleColor];
+}
+
+- (void)updateTitleColor {
+  self.titleLabel.textColor = [self titleColorForState:self.state];
+}
+
+- (void)updateState {
+  [self updateBackgroundColor];
+  [self updateBorderColor];
+  [self updateBorderWidth];
+  [self updateElevation];
+  [self updateTitleColor];
+}
+
+#pragma mark - Control
+
+- (void)setEnabled:(BOOL)enabled {
+  [super setEnabled:enabled];
+
+  [self updateState];
+}
+
+- (void)setHighlighted:(BOOL)highlighted {
+  [super setHighlighted:highlighted];
+
+  [self updateState];
+}
+
+- (void)setSelected:(BOOL)selected {
+  [super setSelected:selected];
+
+  [self updateState];
+  [self setNeedsLayout];
+}
+
+#pragma mark - Layout
+
+- (void)layoutSubviews {
+  _inkView.frame = self.bounds;
+  _imageView.frame = [self imageViewFrame];
+  _selectedImageView.frame = [self selectedImageViewFrame];
+  _accessoryView.frame = [self accessoryViewFrame];
+  _titleLabel.frame = [self titleLabelFrame];
+
+  _selectedImageView.alpha = self.showSelectedImageView ? 1 : 0;
+
+  CGFloat cornerRadius = MIN(CGRectGetHeight(self.frame), CGRectGetWidth(self.frame)) / 2;
+  self.layer.cornerRadius = cornerRadius;
+  self.layer.shadowPath =
+      [UIBezierPath bezierPathWithRoundedRect:self.bounds cornerRadius:cornerRadius].CGPath;
+}
+
+- (CGRect)contentRect {
+  return CGRectMake(self.contentPadding.left,
+                    self.contentPadding.top,
+                    self.bounds.size.width - UIEdgeInsetsHorizontal(self.contentPadding),
+                    self.bounds.size.height - UIEdgeInsetsVertical(self.contentPadding));
```

Comment:
Is this method not just UIInsetsInsetRect() ?

Reply:
It is! Changed to use UIEdgeInsetsInsetRect.